### DISCUSSION TITLE
Blue / Green Deployments via Figmagic?

### DISCUSSION THREAD

**Discussion by nicholasjayantylearns** [mikaelvesavuori/figmagic#201] [General]
I am so so happy to have found this project.  I read the blogs, the articles and I have been thinking about CI/CD tools for UX are one of the missing links.  UX spends hours banging their fist on the table and spills millions of words explaining the value of UX...we need to encode the value of UX into the software delivery pipeline.   Code is the material.  No Industrial designer would manufacture a line of chairs without first understanding the materials and tooling...yet UX does it every day relying on others to translate graphic representations into coded implementations.  - 

When we think of a key UX super power - fast iteration / multiple ideas to test in advance of development to mitigate the risk of developing and productizing specific features - how does figmagic enable multiple iterations of the same design to be A/B tested? Does figmagic make it easier to merge two different design concepts that have elements in one iteration and elements of another together - then push that merged element to the next stage in the software development pipeline?

**Comment by mikaelvesavuori** [comment]
Hi @nicholasjayantylearns, thanks for reaching out here and elsewhere :)

The short and sweet answer to your question about "blue/green" is that Figmagic supports retrieving versions of Figma files. You'd have to design/think up a flow that makes sense on your end, but between the Figmagic + Figma version support and the ability to craft a CI workflow that functionally can get the A data + B data and deploy these independently, then I don't see how Figmagic could not support that need.

Note, again, that dedicated/specific support for this is out of scope for Figmagic, and I don't see how I could make it generic enough, given that it would need to integrate with other tools.
